shotgun_api3/shotgun.py

```diff
@@ -2206,6 +2333,58 @@ def upload(self, entity_type, entity_id, path, field_name=None, display_name=Non
         attachment_id = int(str(result).split(":")[1].split("\n")[0])
         return attachment_id
 
+    def _get_attachment_upload_info(self, is_thumbnail, filename):
+        """
+        Internal function to get the information needed to upload a file to Cloud storage.
+
+        :param bool is_thumbnail: indicates if the attachment is a thumbnail.
+        :param str filename: name of the file that will be uploaded.
+
+        :returns: dictionary containing the ulpload_info and the url to upload to.
+        :rtype: dict
+        """
+
+        if is_thumbnail:
+            upload_type = 'Thumbnail'
+        else:
+            upload_type = 'Attachment'
+
+        params = {
+            "upload_type" : upload_type,
+            "filename" : filename
+        }
+
+        params.update(self._auth_params())
+
+        upload_url = "/upload/api_get_upload_link_info"
+        url = urlparse.urlunparse((self.config.scheme, self.config.server,
+                                   upload_url, None, None, None))
+
+        # Create opener with extended form post support
+        opener = self._build_opener(FormPostHandler)
+
+        # Perform the request
+        try:
+            upload_info = opener.open(url, params).read()
+        except urllib2.HTTPError, e:
+            if e.code == 500:
+                raise ShotgunError("Server encountered an internal error. "
+                                   "\n%s\n(%s)\n%s\n\n" % (url, self._sanitize_auth_params(params), e))
+            else:
+                raise ShotgunError("Unanticipated error occurred uploading "
+                                   "%s: %s" % (path, e))
+        else:
+            if not str(upload_info).startswith("1"):
+                raise ShotgunError("Could not get upload_url but " \
+                                   "not sure why.\nPath: %s\nUrl: %s\nError: %s" % (
+                                       path, url, str(upload_info)))
+
+        LOG.debug("Completed rpc call to %s" % (upload_url))
+        return {
```

Comment:
A comment with an example of the expected return data from sg would be helpful here